packages/cadl-rlc-emitter/src/index.ts

```diff
@@ -1,55 +1,124 @@
-import { CompilerHost, Program } from '@cadl-lang/compiler';
-import { buildClientDefinitions, Paths } from '@azure-tools/rlc-codegen';
-import { getAllRoutes } from '@cadl-lang/rest/http';
-import { dirname, join } from 'path';
+import { CompilerHost, getDoc, ModelType, Program, Type } from "@cadl-lang/compiler";
+import {
+  buildClientDefinitions,
+  Paths,
+  RLCModel,
+  generateSchemaTypes,
+  File,
+  SchemaContext,
+  Schema
+} from "@azure-tools/rlc-codegen";
+import { getAllRoutes } from "@cadl-lang/rest/http";
+import { dirname, isAbsolute, join } from "path";
+import { getSchemaForType } from "./modelUtils.js";
+import { Project } from "ts-morph";
 
 export async function $onEmit(program: Program) {
-    const [routes, _diagnostics] = getAllRoutes(program);
-    const paths: Paths = {};
-    for (const route of routes) {
-        paths[route.path] = {
-            name: route.operation.name,
-            pathParameters: route.parameters.parameters
-                .filter((p) => p.type === 'path')
-                .map((p) => {
-                    return {
-                        name: p.name,
-                        type: 'string',
-                        description: 'param'
-                    };
-                }),
-            methods: {
-                [route.verb]: [
-                    {
-                        description: 'Foo',
-                        hasOptionalOptions: route.parameters.parameters.some(
-                            (p) => p.param.optional
-                        ),
-                        optionsName: 'options',
-                        responseTypes: { success: ['string'], error: [] },
-                        returnType: '',
-                        successStatus: ['200']
-                    }
-                ]
-            }
-        };
+  const rlcModels = await transformRLCModels(program);
+  const project = new Project()
+  await emitCLientDefinition(rlcModels, program);
+  await emitModels(rlcModels, program, project);
+}
+
+async function emitModels(rlcModels: RLCModel, program: Program, project: Project) {
```

Comment:
Do we have any test cases for the current emitter?

Reply:
We have one here https://github.com/Azure/autorest.typescript/blob/cadl-integration/packages/cadl-rlc-test/spec/main.cadl but we don't checkin the generated output file yet.